README.md

```diff
@@ -140,6 +142,106 @@ title,content
 My Title,Hello World!
 ```
 
+### Creating a custom Task
+
+TODO: Add generation instructions
+
+If you have a special use case requiring iteration over an unsupported
+collection type, such as external resources fetched from some API, you can
+implement the `enumerator_builder` method instead.
+
+This method should return an object responding to `enumerator(context:)` with
+an Enumerator, yielding pairs of `[item, item_cursor]` to the `process` method
+your Task defines. In order for your enumerator to support resuming iteration
+part way through, you may use the `context.cursor`.
+
+You may optionally provide an implementation for `count`, if appropriate.
+
+```ruby
+# app/tasks/maintenance/shopping_list_task.rb
+module Maintenance
+  class ShoppingListTask < MaintenanceTasks::Task
+    def enumerator_builder
+      IngredientsEnumerator.new
+    end
+
+    def process(ingredient)
+      ShoppingList.add(ingredient)
+    end
+
+    class IngredientEnumerator
+      def enumerator(context:)
+        Enumerator.new do |yielder|
+          cursor = context.cursor
+
+          if cursor.nil?
+            recipe = FancyRecipeAPI.random_recipe
+            ingredient_id = nil
+          else
+            recipe_id, ingredient_id = cursor.split(':', 2)
```

Comment:
I feel like this example with the two levels of id is a bit too complex. Also a bit weird that the ingredients would be paginated. Can we make it just about a list of recipes being paginated maybe?

Better yet we could use something commerce related in a nod to Shopify? https://shopify.dev/docs/admin-api/rest/reference/orders/order?api[version]=2020-04#index-2020-04 using `Order`, `since_id` and `limit` as the name/params in the example?